language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocument.java

```diff
@@ -72,14 +78,24 @@ public String getContent() {
 
     public void setContent(String content) {
         this.content = content;
+        Path namePart = this.path.getFileName();
         // TODO: Fix this, each time creates a new tree
-        setTree(SyntaxTree.from(TextDocuments.from(this.content)));
+        if (namePart != null) {
+            setTree(SyntaxTree.from(TextDocuments.from(this.content), namePart.toString()));
+        } else {
+            setTree(SyntaxTree.from(TextDocuments.from(this.content)));
+        }
     }
 
     public void setIncrementContent(String content) {
         this.content = content;
         // TODO: Fix this, each time creates a new tree
-        setTree(SyntaxTree.from(TextDocuments.from(this.content)));
+        Path namePath = this.path.getFileName();
+        if (namePath != null) {
+            setTree(SyntaxTree.from(TextDocuments.from(this.content), namePath.toString()));
+        } else {
+            setTree(SyntaxTree.from(TextDocuments.from(this.content)));
+        }
```

Comment:
This code seems to be duplicate. Extract this to a separate method, and use in all places.

Reply:
Cool